--- tiff.c.ORIG	2010-05-14 09:08:25.679384931 +0200
+++ tiff.c	2010-05-14 09:57:01.895807694 +0200
@@ -42,14 +42,14 @@
      struct tiff_info *info;
 {
   ZFILE *zf;
-  TIFFHeader th;
+  TIFFHeaderCommon th;
   TIFF *tiff;
 
   zf = zopen(fullname);
 
   /* read TIFF header and see if it looks right
    */
-  if ((zread(zf, (byte *)&th, sizeof(TIFFHeader)) == sizeof(TIFFHeader)) &&
+  if ((zread(zf, (byte *)&th, sizeof(TIFFHeaderCommon)) == sizeof(TIFFHeaderCommon)) &&
       ((th.tiff_magic == TIFF_BIGENDIAN) ||
        (th.tiff_magic == TIFF_LITTLEENDIAN))) {
 
